### PR TITLE
Doc updates

### LINKED ISSUE #325: postrelease question: what if master branch is protected by a hook?
I've just came across this here:

```
INFO: remote: error: GH006: Protected branch update failed for refs/heads/master.
remote: error: Required status check "continuous-integration/travis-ci" is expected.
To github.com:collective/plone.formwidget.geolocation.git
! [remote rejected] master -> master (protected branch hook declined)
```

postrelease cannot push changes to repository if it's protected by a hook ... what can one do in that case?

### PR DESCRIPTION
**Pull request by htgoebel**
- Update documentation about entry-points (closes #325)
- Add section about how to build the documentation locally.
- Use sphinx_rtd_theme when building docs locally.

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -72,3 +73,4 @@ recipe = collective.recipe.sphinxbuilder
 source = ${buildout:directory}/doc/source
 build = ${buildout:directory}/doc/build
 eggs = zest.releaser
+ sphinx_rtd_theme
```
Comment (reinout) [inline buildout.cfg]: Indentation tweak:

    eggs =
        zest.releaser
        sphinx_rtd_theme

File: doc/source/developing.rst [hunk 2]
```diff
@@ -72,3 +72,20 @@ And the same for git, so you should do::
...
+Building the documentation locally
+-------------------------------------
+
+If you worked on the documentation, we suggest you verify the markup
+and the result by building the documentation locally and view your
+results.
+
+For building the documentation simply run::
+
+    bin/sphinx
```
Comment (reinout) [inline doc/source/developing.rst]: Add a $ to indicate a shell command:

    $ bin/sphinx

(Like in developing.rst)

File: doc/source/developing.rst [hunk 3]
```diff
@@ -72,3 +72,20 @@ And the same for git, so you should do::
...
+
+For viewing the documentation open :file:`doc/build/html/index.html`
+in your browser, e.g. by running::
+
+    xdg-open doc/build/html/index.html
```
Comment (reinout) [inline doc/source/developing.rst]: Same "$"

**Comment by htgoebel** [comment]
@reinout Updated this one.

**Pull request merged**